### DISCUSSION TITLE
Config to include Security / Vulnerability in grouping.

### DISCUSSION THREAD

**Discussion by tmossboss** [renovatebot/renovate#15107] [Request Help]
Currently using the following config to group all non major patches into one PR. 
```json
{
    "packageRules": [
      {
        "matchPackagePatterns": ["*"],
        "matchUpdateTypes": [
          "minor",
          "patch",
          "patch",
          "pin",
          "digest",
          "lockFileMaintenance",
          "rollback",
          "bump"
        ],
        "groupName": "all non-major dependencies",
        "groupSlug": "all-minor-patch",
        "schedule": ["every weekend"]
      }
    ],
    "vulnerabilityAlerts": {
      "labels": ["security"]
    },
    "labels": ["dependencies"]
}
  ```
  
  This works great but currently Renovate also creates separate PR's specifically for anything marked as a SECURITY issue. I want these included in my grouping. I am using the * wildcard os I am not sure if this is possible atm.
  
  Edit for clarity, all of the security patches are minor version changes not major.

**Comment by rarkins** [comment]
You might find some success by altering the `vulnerabilityAlerts` object's default values: https://github.com/renovatebot/renovate/blob/5c31446599297e5006dd39aee3a07a3c952aaa01/lib/config/options/index.ts#L1476-L1488

**Reply by richardkazuomiller** [reply]
It would be nice to be able to override the configuration so that vulnerability alert PRs are included in the group defined in `packageRules` but since `vulnerabilityAlerts` is mergeable, it looks like there is no way to override the config to make the group stay together. I ran into this with `semantic-release` having a major update which is also a security update that can't be merged due to incompatible `@semantic-release/*` peer dependences.

**Comment by hayawata3626** [comment]
@rarkins 
I would like to do the same thing.

> This works great but currently Renovate also creates separate PR's specifically for anything marked as a SECURITY issue. I want these included in my grouping. I am using the * wildcard os I am not sure if this is possible atm.


What I have tried.
I was able to combine the PRs into one by setting the vulnerabilityAlerts groupName: "vulnerabilityAlerts"`. But I can't figure out how to separate PRs like major, minor, patch version, etc. by looking at the vulnerabilityAlerts settings.

renovate.json
```json
{
  "extends": [
    "config:base",
    ":timezone(Asia/Tokyo)",
    ":masterIssue",
    "group:allNonMajor"
  ],
  "vulnerabilityAlerts": {
    "groupName": "vulnerabilityAlerts",
    "enabled": true,
    "schedule": [],
    "prCreation": "immediate",
    "labels": ["renovate"]
  },
  "osvVulnerabilityAlerts": true,
  "transitiveRemediation": true,
  "baseBranches": ["main"]
}

```

**Comment by tdanecker** [comment]
I also wanted to group all minor and patch updates of vulnerable dependencies together and create separate PRs for major updates. After some tries, I found this solution:

```json
  "vulnerabilityAlerts": {
    "enabled": true,
    "groupName": "vulnerable dependencies",
    "packageRules": [
      {
        "matchManagers": ["npm"],
        "matchUpdateTypes": ["major"],
        "additionalBranchPrefix": "{{{depNameSanitized}}}-"
      }
    ]
  }
```

The `vulnerabilityAlerts` configuration (incl. it's default) overwrites any configs from `packageRules` (internally via `force`). This especially includes the `groupName` and `branchTopic`. The only config I found which is not forced by `vulnerabilityAlerts` was the `additionalBranchPrefix`. As renovate in the end groups updates according to branch names, setting the `additionalBranchPrefix` splits the major updates apart. But don't forget to also set the `groupName` in the `vulnerabilityAlerts` for this to work.
